Lib/test/test_concurrent_futures/test_process_pool.py

```diff
@@ -228,59 +228,77 @@ def mock_start_new_thread(func, *args, **kwargs):
+    def test_process_pool_executor_terminate_kill_workers(self):
+        for function_name in ('terminate_workers', 'kill_workers'):
+            manager = multiprocessing.Manager()
+            q = manager.Queue()
 
-        with futures.ProcessPoolExecutor(max_workers=1) as executor:
-            executor.submit(_put_sleep_put, q)
+            with futures.ProcessPoolExecutor(max_workers=1) as executor:
+                executor.submit(_put_sleep_put, q)
 
-            # We should get started, but not finished since we'll terminate the workers just after
-            self.assertEqual(q.get(timeout=1), 'started')
+                # We should get started, but not finished since we'll terminate the workers just after
+                self.assertEqual(q.get(timeout=1), 'started')
 
-            executor.terminate_workers()
+                worker_process = list(executor._processes.values())[0]
+                getattr(executor, function_name)()
+                worker_process.join()
 
-            self.assertRaises(queue.Empty, q.get, timeout=1)
+                if function_name == 'terminate_workers' or sys.platform == 'win32':
+                    # On windows, kill and terminate both send SIGTERM
+                    self.assertEqual(worker_process.exitcode, -signal.SIGTERM)
+                elif function_name == 'kill_workers':
+                    self.assertEqual(worker_process.exitcode, -signal.SIGKILL)
 
-    def test_process_pool_executor_terminate_workers_dead_workers(self):
-        with futures.ProcessPoolExecutor(max_workers=1) as executor:
-            future = executor.submit(os._exit, 1)
-            self.assertRaises(BrokenProcessPool, future.result)
+                self.assertRaises(queue.Empty, q.get, timeout=1)
 
-    @unittest.mock.patch('concurrent.futures.process.os.kill')
-    def test_process_pool_executor_terminate_workers_not_started_yet(self, mock_kill):
-        with futures.ProcessPoolExecutor(max_workers=1) as executor:
-            # The worker has not been started yet, terminate_workers should basically no-op
-            executor.terminate_workers()
+    def test_process_pool_executor_terminate_kill_workers_dead_workers(self):
+        for function_name in ('terminate_workers', 'kill_workers'):
+            with futures.ProcessPoolExecutor(max_workers=1) as executor:
+                future = executor.submit(os._exit, 1)
+                self.assertRaises(BrokenProcessPool, future.result)
 
-        mock_kill.assert_not_called()
+                # even though the pool is broken, this shouldn't raise
+                getattr(executor, function_name)()
 
-    def test_process_pool_executor_terminate_workers_stops_pool(self):
-        with futures.ProcessPoolExecutor(max_workers=1) as executor:
-            executor.submit(time.sleep, 0).result()
+    def test_process_pool_executor_terminate_kill_workers_not_started_yet(self):
+        for function_name in ('terminate_workers', 'kill_workers'):
```

Comment:
in all of these tests, using `with self.subTest(function_name):` would make for nicer test results.  or refactoring these into _test_blah(self, function_name): helper functions that top level test_blah_terminate and test_blah_kill call so they're distinct actual test cases.

Reply:
Moments like this wish we had pytest to do `pytest.mark.parametrize`. 

I'll swap over to using subTest in the next iteration. Thanks!